### ISSUE TITLE
[feat]: <rh-tabs> ready to migrate

### ISSUE THREAD

**Issue by markcaron** [RedHat-UX/red-hat-design-system#537]
### Tag Name

pfe-tabs

### RHDS Only

- [ ] The component will be **RHDS only**, delete the PFE component once published

### Additional Info

Extend/build `<rh-tabs>` component from `<pfe-tabs>`.

[Current design specs](https://ux.redhat.com/components/tabs/)

**Comment by coreyvickery** [comment]
@zeroedin Here are the specs for `rh-tabs`. There are a few very minor changes we need to make to `pfe-tabs` as part of moving to `rh-tabs`, just let me know where to add them.

https://xd.adobe.com/spec/3e62e93c-8338-4f31-5040-3b81f0ed5c71-4853/screen/6b20a57b-62d0-4973-ab9f-a10c004eabfe/specs/

**Comment by zeroedin** [comment]
Updated Specs including panel spacing: https://xd.adobe.com/spec/7d458cbb-fd45-4988-a542-4a94c9a3ecbb-2816/

**Comment by marionnegp** [comment]
@coreyvickery @zeroedin, what do you think about [these changes](https://xd.adobe.com/view/f9ac9c3d-f8c2-47b1-b329-19541c6ba404-acff/) to the horizontal overflow tabs on mobile screens only?


![Image](https://user-images.githubusercontent.com/95588923/214318066-bb6b5421-ee2a-4ba7-a8d7-25e53dd2173c.png)

**Comment by marionnegp** [comment]
Updates to spacing for tab panel and tabs with insets: [XD link](https://xd.adobe.com/view/03de5835-dc57-4da5-8328-18b0305be95a-568d/)

**Comment by marionnegp** [comment]
@markcaron @coreyvickery, since we don't have container queries yet, @zeroedin and I were talking about how tab panel padding should behave in smaller containers. For now, I thought that keeping the panel content aligned with the overflow button borders everywhere might be enough for an MVP version of `rh-tabs`. 

I made some mockups showing how that would look if `rh-tabs` were used multiple times on one page: [XD link](https://xd.adobe.com/view/d8893fae-33a7-42ac-b32b-2387f17a927a-8dda/). Do you think this makes sense? In a future state, we'd keep the panel content aligned with the tab text, unless the container got too small.

**Comment by markcaron** [comment]
Between slides 1 and 3, it seems like there's a different proposed padding when the overflow arrows kick in.

Slide 1: aligned with border edge of arrow
<img width="677" alt="image" src="https://user-images.githubusercontent.com/1890950/218847648-07fd6971-dd28-45a3-a641-81318118e6ec.png">

Slide 3: aligned with text in tab
<img width="665" alt="image" src="https://user-images.githubusercontent.com/1890950/218847828-15df4872-d288-4ab7-92e5-534c4747d3f5.png">

IMO, I prefer Slide 1's approach with aligning with the border edge. Aligning with the text in overflow mode is just a bit too far, plus there will be inconsistencies when the tabs are actually scrolled (they won't always line up except on the first tab).

When not in overflow mode, lining up with the text makes sense.
<img width="478" alt="image" src="https://user-images.githubusercontent.com/1890950/218848237-1ab7783e-4965-4da5-8e7a-c63fb7b49dba.png">

As for the smaller screen, it feels like the line length is too narrow, and we could probably get away with aligning the padding/text with the outer edge of the arrow icon itself.
<img width="328" alt="image" src="https://user-images.githubusercontent.com/1890950/218849274-fb4548bf-8840-44d0-886d-63af75171388.png">

My $0.02 anyway.

**Comment by marionnegp** [comment]
True, it was fine until I tried scrolling to the very last tab. Updated the [same link](https://xd.adobe.com/view/d8893fae-33a7-42ac-b32b-2387f17a927a-8dda/) to eliminate the third screen and update padding on mobile. @zeroedin, will this work?
 
![Screenshot 2023-02-14 at 3 25 43 PM](https://user-images.githubusercontent.com/95588923/218854051-4e902792-6205-4b1b-9628-053c89c117d3.png)

**Comment by markcaron** [comment]
@marionnegp that looks a lot nicer, IMO! Especially the smaller screen.

I think it'll be easier to control, too, since there will be so many variables between the arrow width/padding and the tab padding.

**Comment by bennypowers** [comment]
we can simulate container queries using a controller

the controller's internal implementation can short circuit if it detects that container size queries are supported, as in this pseudocode
```css
#container {
  container: inline-size;
}
```
```ts
#container = new ContainerQueryController(this, {
  selector: '#container',
  // maybe we can automatically get this from the host's adoptedStylesheets
  // probably can't support the new units though, at least not in a first pass
  query: '(min-width: 200px)',
});
```
```ts
class ContainerQueryController {
  static #supports = CSS.supports('container-type: inline-size');

  #ro?: ResizeObserver;
  
  matches: boolean;
  
  constructor() {
    if (!ContainerQueryController.#supports) {
      this.#ro = new ResizeObserver(...);
    }
  }
}
```
check if container size queries are supported, and if so use `matches` to return it's state
